pcdsdevices/slits.py

```diff
@@ -0,0 +1,321 @@
+    height. This means that if you have a nominal aperture of 2 mm, but your
+    slits are set to 0.5 mm, the total estimated transmission will be 25%.
+    Obviously this is greatly oversimplified, but it allows the lightpath to
+    make a rough back of the hand calculation without being over aggressive
+    about changing slit widths during alignment
+    """
+    xcenter = C(SlitPositioner, '', slit_type="XCENTER")
+    xwidth = C(SlitPositioner, '', slit_type="XWIDTH")
+    ycenter = C(SlitPositioner, '', slit_type="YCENTER")
+    ywidth = C(SlitPositioner, '', slit_type="YWIDTH")
+    blocked = C(EpicsSignalRO, ":BLOCKED")
+    open_cmd = C(EpicsSignal, ":OPEN")
+    close_cmd = C(EpicsSignal, ":CLOSE")
+    block_cmd = C(EpicsSignal, ":BLOCK")
+
+    # Subscription information
+    SUB_STATE = 'sub_state_changed'
+    _default_sub = SUB_STATE
+
+    # Default Attributes
+    _default_read_attrs = ['xwidth', 'ywidth']
+    _default_configuration_attrs = ['xcenter.readback', 'ycenter.readback']
+
+    def __init__(self, *args, nominal_aperture=(5.0, 5.0), **kwargs):
+        self._has_subscribed = False
+        self.nominal_aperture = nominal_aperture
+        super().__init__(*args, **kwargs)
+
+    def move(self, size, wait=False, moved_cb=None, timeout=None):
+        """
+        Set the dimensions of the width/height of the gap to width paramater.
+
+        Parameters
+        ---------
+        size : float, tuple
+            Target size for slits in both x and y axis. Either specify as a
+            tuple for a rectangular aperture (width, height) or set both with
+            single floating point value to use set a square width
+
+        wait : bool
+            If true, block until move is completed
+
+
+        timeout: float, optional
+            Maximum time for the motion. If None is given, the default value of
+            `xwidth` and `ywidth` positioners is used.
+
+        moved_cb: callable, optional
+            Function to be run when the operation finishes. This callback
+            should not expect any arguments or keywords
+
+        Returns
+        -------
+        status : AndStatus
+            Logical combination of the request to both horizontal and vertical
+            motors
+        """
+        # Check for rectangular setpoint
+        if isinstance(size, tuple):
+            (width, height) = size
+        else:
+            width, height = size, size
+        # Instruct both width and height then combine the output status
+        x_stat = self.xwidth.move(width, wait=False, timeout=timeout)
+        y_stat = self.ywidth.move(height, wait=False, timeout=timeout)
+        status = x_stat & y_stat
+        # Add our callback if one was given
+        if moved_cb is not None:
+            status.add_callback(moved_cb)
+        # Wait if instructed to do so. Stop the motors if interrupted
+        if wait:
+            try:
+                status_wait(status)
+            except KeyboardInterrupt:
```

Comment:
Are we going to be adding logic like this into every class's `move` method? This seems like another thing that will get repetitive that should have some shared code, though it isn't obvious to me how to share it.

Reply:
I think this one is special because of the combined statuses. Hopefully we can use the built-in `move` method associated with `PVPositioner`